components/nimbus/src/tests/test_behavior.rs

```diff
@@ -214,7 +214,10 @@ mod single_interval_counter_tests {
 
     #[test]
     fn test_advance_do_not_advance() -> Result<()> {
-        let mut counter = SingleIntervalCounter::new(IntervalConfig::new(7, Interval::Days));
+        let mut counter = SingleIntervalCounter::from(
+            IntervalData::from(VecDeque::new(), 7, Utc::now()),
+            IntervalConfig::new(7, Interval::Days),
+        );
```

Comment:
Rather than implementing `from` for these structs, which might be confused with `impl From`, would struct initializer syntax work here? Something like:

```suggestion
        let mut counter = SingleIntervalCounter {
            IntervalData {
                [0].into(),   // or VecDeque::from([0]) 
                7, 
                Utc::now()
            },
            IntervalConfig::new(7, Interval::Days),
        };
```

Reply:
Sure, I can remove those methods!